ui/src/app/settings/components/GeneralForm/GeneralForm.js

```diff
@@ -0,0 +1,90 @@
+import { Formik } from "formik";
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import * as Yup from "yup";
+
+import actions from "app/settings/actions";
+import selectors from "app/settings/selectors";
+import Button from "app/base/components/Button";
+import Form from "app/base/components/Form";
+import Input from "app/base/components/Input";
+import Loader from "app/base/components/Loader";
+import Row from "app/base/components/Row";
+import Col from "app/base/components/Col";
+
+const GeneralSchema = Yup.object().shape({
+  maas_name: Yup.string().required(),
+  enable_analytics: Yup.boolean()
+});
+
+const GeneralForm = () => {
+  const dispatch = useDispatch();
+  const saveConfig = actions.config.save;
+
+  const maasName = useSelector(selectors.config.maasName);
+  const enableAnalytics = useSelector(selectors.config.enableAnalytics);
+
+  return (
+    <Row>
+      <Col size="6">
+        <Formik
+          initialValues={{
+            maas_name: maasName,
+            enable_analytics: enableAnalytics
+          }}
+          validationSchema={GeneralSchema}
+          onSubmit={(values, { setSubmitting }) => {
+            setTimeout(() => {
+              Object.keys(values).forEach(key => {
+                dispatch(saveConfig(key, values[key]));
+              });
+              setSubmitting(false);
+            }, 500);
+          }}
+        >
+          {({
```

Comment:
You could consider moving this into a component, see: https://github.com/canonical-web-and-design/maas-ui/pull/62#discussion_r312313966.

Reply:
I've tried following this pattern but keep running into issues. Seems simpler to keep it as it is.

Reply:
What is the benefit of using the formik field?

It seems to wrap an input component which then wraps the input in a field component? That seems like a lot just to output an input.

Also why do we need to pass the data for all fields into each formik field?

Reply:
> I've tried following this pattern but keep running into issues. Seems simpler to keep it as it is.

Please let us know what those issues are, we can't help resolve them otherwise.

Reply:
> What is the benefit of using the formik field?
> 
> It seems to wrap an input component which then wraps the input in a field component? That seems like a lot just to output an input.
> 
> Also why do we need to pass the data for all fields into each formik field?

Management of form state, please read the formik docs.

Reply:
But why can't we do that directly in the input component? The formik wrapper seems like an unnecessary layer of abstraction.

Reply:
Having a look at the `FormikField` component, my answer would be to reduce duplication, given every field needs the same handlers and error object from formik props. @huwshimi may have more to add.